src/com/palmergames/bukkit/util/SupportUtil.java

```diff
@@ -0,0 +1,185 @@
+package com.palmergames.bukkit.util;
+
+import org.bukkit.Bukkit;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+
+/**
+ * Allows testing support for known unsupported classes
+ * @author GNosii
+ */
+public class SupportUtil {
+
+	private static final HashMap<String, Support> TESTS = getSupportData();
```

Comment:
Any hashmap fields/returns can just be replaced with Map, the implementation used matters very little in this case.